plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/function/tablechanges/DeltaLakeTableChangesFunction.java

```diff
@@ -0,0 +1,190 @@
+
+        int indexOfDot = schemaTableName.indexOf(".");
+        String schemaName = schemaTableName.substring(0, indexOfDot);
+        String tableName = schemaTableName.substring(indexOfDot + 1);
+        MetadataEntry metadata = getTableMetadata(session, schemaName, tableName);
+
+        ImmutableList.Builder<Descriptor.Field> outputFields = ImmutableList.builder();
+        List<DeltaLakeColumnMetadata> deltaLakeColumnMetadata = extractSchema(metadata, typeManager);
+        deltaLakeColumnMetadata.stream()
+                .map(column -> new Descriptor.Field(column.getName(), Optional.of(column.getType())))
+                .forEach(outputFields::add);
+        outputFields.add(new Descriptor.Field(CHANGE_TYPE_COLUMN_NAME, Optional.of(VARCHAR)));
+        outputFields.add(new Descriptor.Field(COMMIT_VERSION_COLUMN_NAME, Optional.of(INTEGER)));
+        outputFields.add(new Descriptor.Field(COMMIT_TIMESTAMP_COLUMN_NAME, Optional.of(TIMESTAMP_TZ_MILLIS)));
+        List<Descriptor.Field> requiredColumns = outputFields.build();
+
+        List<DeltaLakeColumnHandle> columns = deltaLakeColumnMetadata.stream()
+                .map(column -> toColumnHandle(
+                        column.getColumnMetadata(),
+                        column.getFieldId(),
+                        column.getPhysicalName(),
+                        column.getPhysicalColumnType(),
+                        metadata.getCanonicalPartitionColumns()))
+                .collect(toImmutableList());
+
+        return TableFunctionAnalysis.builder()
+                .handle(new TableChangesFunctionTableHandle(catalogHandle, schemaName, tableName, startVersion, endVersion, columns))
+                .returnedType(new Descriptor(requiredColumns))
+                .build();
+    }
+
+    private MetadataEntry getTableMetadata(ConnectorSession session, String schemaName, String tableName)
+    {
+        HiveMetastoreBackedDeltaLakeMetastore metastore = new HiveMetastoreBackedDeltaLakeMetastore(
+                hiveMetastoreFactory.createMetastore(Optional.of(session.getIdentity())),
+                transactionLogAccess,
+                typeManager,
+                statisticsAccess,
+                fileSystemFactory);
+
+        Optional<Table> table = metastore.getTable(schemaName, tableName);
+        if (table.isEmpty()) {
+            throw new IllegalArgumentException(schemaName + "." + tableName + "doesn't exist");
+        }
+        SchemaTableName schemaTableName = new SchemaTableName(schemaName, tableName);
+        TableSnapshot tableSnapshot = metastore.getSnapshot(schemaTableName, session);
+        Optional<MetadataEntry> metadata = metastore.getMetadata(tableSnapshot, session);
+        if (metadata.isEmpty()) {
+            throw new TrinoException(NOT_FOUND, "metadata for " + schemaTableName + " is not available");
+        }
+        return metadata.get();
+    }
+
+    public enum FileSource
```

Comment:
I'd probably give this its own file rather than nesting it inside of `TableChangesFunction`. You already have the `tablechanges` package to scope it